src/v/cluster/scheduling/partition_allocator.cc

```diff
@@ -81,6 +81,92 @@ allocation_constraints partition_allocator::default_constraints() {
     return req;
 }
 
+bool guess_is_memory_group_aware_memory_limit() {
+    // Here we are trying to guess whether `topic_memory_per_partition` is
+    // memory group aware. Originally the property was more like a guess of how
```

Comment:
I don't think it's true. It was a empirical measurement to find an X where 8 GB / X gave a number of partitions that worked on 8 GB instances and 4 GB / X gave a number that worked on 4 GB instances, based on testing on those two instance types (which only really happened incidentially because Arm had some 4 GB instance types we wanted to try).

It was never a guess at the static memory usage of a partition: it was entirely driven by "OOM or not" during tests so included also dynamic memory use which "should" be accounted for in other places (but it it is not always the case).

Just so the history makes some sense (this all makes more sense given the info it was 1 MB not 10 MB: the problem which was noted is that on 4 GB hosts you could not reach the partitios_per_shard value before OOMing, thus this memory setting needed to be raised to lower the limit in that case).

Reply:
changed